meta-facebook/yv35-cl/src/platform/plat_sensor_table.c

```diff
@@ -242,6 +252,44 @@ void pal_fix_sensor_config()
 			}
 			break;
 		case SYS_BOARD_EVT3_HOTSWAP:
+			/* Follow the GPIO table, the HSC device type can be by ADC7(net name: HSC_TYPE_ADC)
+			 * If the voltage of ADC-7 is 0.5V(+/- 15%), the hotswap model is ADM1278.
+			 * If the voltage of ADC-7 is 1.0V(+/- 15%), the hotswap model is LTC4282.
+			 * If the voltage of ADC-7 is 1.5V(+/- 15%), the hotswap model is LTC4286.
+			 */
+			voltage_hsc_type_adc = get_hsc_type_adc_voltage();
+			if ((voltage_hsc_type_adc > 0.5 - (0.5 * 0.15)) &&
+			    (voltage_hsc_type_adc < 0.5 + (0.5 * 0.15))) {
+				printf("Added ADM1278 sensor configuration\n");
+				sensor_count = ARRAY_SIZE(adm1278_sensor_config_table);
+				while (sensor_count != 0) {
+					add_sensor_config(
+						adm1278_sensor_config_table[--sensor_count]);
+					if (sensor_count == 0) {
+						break;
+					}
+				}
+			} else if ((voltage_hsc_type_adc > 1.0 - (1.0 * 0.15)) &&
+					   (voltage_hsc_type_adc < 1.0 + (1.0 * 0.15))) {
+				printf("TODO: Support LTC4282 sensor config\n");
+			} else if ((voltage_hsc_type_adc > 1.5 - (1.5 * 0.15)) &&
+				   (voltage_hsc_type_adc < 1.5 + (1.5 * 0.15))) {
+				printf("TODO: Support LTC4286 sensor config\n");
+			} else {
+				printf("Unknown hotswap model type, HSC_TYPE_ADC voltage: %fV\n",
+				       voltage_hsc_type_adc);
+			}
+			/* Replace the temperature sensors configuration including "HSC Temp" and "MB Outlet Temp."
+			 * For these two sensors, the reading values are read from TMP431 chip.data.num
+			 */
+			sensor_count = ARRAY_SIZE(evt3_class1_adi_temperature_sensor_table);
+			while (sensor_count != 0) {
+				add_sensor_config(
+					evt3_class1_adi_temperature_sensor_table[--sensor_count]);
+				if (sensor_count == 0) {
+					break;
+				}
+			}
```

Comment:
This seems like it might be better as a for loop?

Reply:
We have fixed this and do the forced update.
Please review it. Thanks.